A-4E-C/Cockpit/Scripts/Systems/weapon_system.lua

```diff
@@ -558,8 +558,12 @@ function update()
 
 
         for i = 1, num_stations do
-            if station_states[i] == STATION_READY then
-                local station_info = WeaponSystem:get_station_info(i-1)
+            local station_info = WeaponSystem:get_station_info(i-1)
```

Comment:
I think an alternative is to save the trigger_state at the start of this big if block. As we know at least the bomb-release or the trigger are pressed. So if we save trigger_engaged into a variable on line 536 and then use that in the check on line 601 instead of trigger_engaged, that would account for the special scenario involving the gunpods without making any changes to existing weapons.

Reply:
Yes, I just noticed the ripple_sequence_end does more than just releasing trigger. And that also needed for Rocket AWRS to work.
I think i got your idea, store if trigger_engaged was pressed at start of  block, allow AWRS logic do what it needs, set everything off and at the gunpod fire check if trigger was pressed before AWRS logic and fire the gunpod at that cycle anyway.
But i was thinking maybe the best approach would be removing the gunpod firing loop outside the station ready fire loop. Since its logic to fire does not depend on Station Readiness or the AWRS logic in anyway, what do you think?

Does AWRS interact with Gunpods in anyway? I mean firing tone or the valid_solution from target computer in anyway?
if it doesn't, We could have a first internal loop for the Gunpod logic that would only run if Gunpods attached, and another for all other station types.
That way if no Gunpods attached that first loop would not run, and save some performance ticks as well. Not sure how performance for the code in Lua in DCS impacts cpu or game performance much or not.

Many questions in my head, right now.

Reply:
Extracting the gunpods from the loop would actually require you to have another loop over the pylons that would execute in addition to this one. This would be more of a performance hit, however as we know **premature optimisation is the root of all evil** and it's actually far more important that this code is maintainable (or at least more maintainable than it is haha) especially when it will essentially be negligible.

The gunpods are essentially a separate system, hence the weird way of enabling them, they don't interact with the AWRS.

If you were to take that approach I would extract that logic into a function (for maintainability), calling your function from before where the trigger gets set to off. I'll leave it to you. Let me know if you have any more questions.

Reply:
Got it, moving the Gunpod fire logic above the trigger release also crossed my mind, I´ll give it another go and update the PR.

Reply:
@08jne01 I found a couple of issues, first was the trigger release problem. Hard to find a fix that does not break rocket AWRS when using trigger, partial fix was only release trigger if mode = ROCKET since from what i understand rockets are only weapon that uses AWRS with trigger release.
Then I had to skip the release_max check break if station is a gunpod.
That is a partial fix cause guns will work now on all modes besides CMPTR cause valid_solution is most of the time false on that function and that makes not go into the firing loop.

Questions: Guns stoping firing at the same time as rockets salvo ends on rocket mode acceptable?
Because that fix the guns not firing at all on the other modes besides CMPTR

To fix the guns not firing on CMPTR mode, we would have to move all the gunpod logic outside the main firing loop, or what i was thinking to change the if conditions on the main loop to move the cmptr check to inside the main loop.

Reply:
I'm not a fan of this really specific logic to let it through in certain conditions. I think this would be much cleaner in a single function say called fire gunpods. Where it checks if there are gunpods on the pylons and fires them if the trigger is depressed. This way you can bypass the computer mode too. This would be called before the other logic is executed, so it bypasses the check which disables the trigger. Similar to how the smokes are called.